core/chains/evm/logpoller/log_poller.go

```diff
@@ -1023,8 +1023,16 @@ func (lp *logPoller) latestBlocks(ctx context.Context) (*evmtypes.Head, int64, e
 		return nil, 0, fmt.Errorf("failed to get latest and latest finalized block from HeadTracker: %w", err)
 	}
 
-	lp.lggr.Debugw("Latest blocks read from chain", "latest", latest.Number, "finalized", finalized.BlockNumber())
-	return latest, finalized.BlockNumber(), nil
+	finalizedBN := finalized.BlockNumber()
+	// This is a dirty trick that allows LogPoller to function properly in tests where chain needs significant time to
+	// reach finality depth. An alternative to this one-liner is a database migration that drops restriction
+	// LogPollerBlock.FinalizedBlockNumber > 0 (which we actually want to keep to spot cases when FinalizedBlockNumber was simply not populated)
+	// and refactoring of queries that assume that restriction still holds.
+	if finalizedBN == 0 {
+		finalizedBN = 1
+	}
+	lp.lggr.Debugw("Latest blocks read from chain", "latest", latest.Number, "finalized", finalizedBN)
+	return latest, finalizedBN, nil
```

Comment:
In the past, we've just required all tests to make sure the chain has reached finality before running. Is there a test where this is infeasible?

I don't think we should make this change as-is without changing anything else, because it will lead people to assume it means that logpoller fully supports chains shorter than finality depth.  If we do add support for it, we'd need to change some other things and remove the comment saying it's not supported:
https://github.com/smartcontractkit/chainlink/pull/14366/files#diff-08d58e904d709d1cb1a2ba1e99cce7dfd594bbf7369f05d3b187e10316dc5babL606-L612

Reply:
> In the past, we've just required all tests to make sure the chain has reached finality before running. Is there a test where this is infeasible?
CCIP load tests do not follow this requirement. Also it's not feasible on chains with large finality depth as it might take hours for test to start generating load.

> If we're adding support for chains shorter than finality depth, then we should remove this comment saying it's not supported:
Nice catch. Removed

> Also, are you sure there aren't other places where we make that assumption... does this change fix it everywhere?
Yes, this seems to be the only place where LP makes assumptions on min chain length.